lib/luanetfilter.c

```diff
@@ -0,0 +1,122 @@
+* a copy of this software and associated documentation files (the
+* "Software"), to deal in the Software without restriction, including
+* without limitation the rights to use, copy, modify, merge, publish,
+* distribute, sublicense, and/or sell copies of the Software, and to
+* permit persons to whom the Software is furnished to do so, subject to
+* the following conditions:
+*
+* The above copyright notice and this permission notice shall be
+* included in all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+#include <linux/module.h>
+#include <linux/version.h>
+#include <linux/netfilter.h>
+#include <linux/netfilter/x_tables.h>
+
+#include <lunatik.h>
+
+typedef struct luanetfilter_xtable_s {
+	lunatik_object_t *runtime;
+} luanetfilter_xtable_t;
+
+static int luanetfilter_xtable(lua_State *L);
+
+static void luanetfilter_release(void *private)
+{
+	luanetfilter_xtable_t *luaxtable = (luanetfilter_xtable_t *)private;
+	lunatik_putobject(luaxtable->runtime);
+}
+
+static const lunatik_reg_t luanetfilter_protocol[] = {
+	{"UNSPEC", NFPROTO_UNSPEC},
+	{"INET", NFPROTO_INET},
+	{"IPV4", NFPROTO_IPV4},
+	{"IPV6", NFPROTO_IPV6},
+	{"ARP", NFPROTO_ARP},
+	{"NETDEV", NFPROTO_NETDEV},
+	{"BRIDGE", NFPROTO_BRIDGE},
+	{NULL, 0}
+};
+
+static const lunatik_reg_t luanetfilter_action[] = {
+	{"DROP", NF_DROP},
+	{"ACCEPT", NF_ACCEPT},
+	{"STOLEN", NF_STOLEN},
+	{"QUEUE", NF_QUEUE},
+	{"REPEAT", NF_REPEAT},
+	{"STOP", NF_STOP},
+	{"CONTINUE", XT_CONTINUE},
+	{"RETURN", XT_RETURN},
+	{NULL, 0}
+};
+
+static const lunatik_namespace_t luanetfilter_flags[] = {
+	{"action", luanetfilter_action},
+	{"proto", luanetfilter_protocol},
+	{NULL, NULL}
+};
+
+static const luaL_Reg luanetfilter_lib[] = {
+	{"xtable", luanetfilter_xtable},
+	{NULL, NULL}
+};
+
+static const luaL_Reg luanetfilter_mt[] = {
+	{"__gc", lunatik_deleteobject},
+	{NULL, NULL}
+};
+
+static const lunatik_class_t luanetfilter_class = {
+	.name = "netfilter",
+	.methods = luanetfilter_mt,
+	.release = luanetfilter_release,
+	.sleep = false,
+	.pointer = true,
+};
+
+static int luanetfilter_xtable(lua_State *L)
+{
+	lunatik_object_t *object;
+	luanetfilter_xtable_t *luaxtable;
+
+	object = lunatik_newobject(L, &luanetfilter_class, sizeof(luanetfilter_xtable_t));
+	luaxtable = (luanetfilter_xtable_t *)object->private;
+
+	memset(luaxtable, 0, sizeof(luanetfilter_xtable_t));
+
+	luaxtable->runtime = lunatik_toruntime(L);
+	lunatik_getobject(luaxtable->runtime);
+
```

Comment:
sorry to be picky here, but we avoid this new line before `return` on the other libraries.. but I will leave the decision to you.. (just saying =)..